### ISSUE TITLE
Why are environment variables CaSE SEnsItiVE?

### ISSUE THREAD

**Issue by szdavid92** [mapbox/node-pre-gyp#280]
Detecting and parsing environment variables should be case insensitive. 
The reason is that Windows treats them so, and the people are pretty used to this behaviour.
I am used to set env vars in upper case so I expected `export NODE_PRE_GYP_BUCKET=mybucket` to work. I am happy to submit a PR for this.

**Comment by springmeyer** [comment]
@szdavid92 👍 I agree that making the support for variables case insensitive would be reasonable. Thanks for offering to provide a PR. I would be happy to review it.

As far as which variables this would impact, and how it would be supported:

  - the two key ones node-pre-gyp respects are for aws keys for publishing https://github.com/mapbox/node-pre-gyp#3-configure-aws-credentials.
   - node-pre-gyp currently uses `rc` which handles pulling environment variables out of the process.env: https://github.com/dominictarr/rc

**Comment by springmeyer** [comment]
> Why are environment variables CaSE SEnsItiVE?

To add more detail: I think they are case sensitive because of how https://github.com/dominictarr/rc. But that is worth a check - I'm not positive, been a while since I dug into the usage of `rc`.

**Comment by szdavid92** [comment]
Do we convert `process.env` to lowercase before [this step](https://github.com/mapbox/node-pre-gyp/blob/master/lib/node-pre-gyp.js#L134)? This is a case sensitive match, so `NODE_PRE_GYP_SomeThing`s won't be added on the config object.

**Comment by szdavid92** [comment]
Sorry, my mistake, those are `npm_config_` variables.

**Comment by szdavid92** [comment]
Yes you are right that `rc` is case sensitive as well. There is already a [PR for this](https://github.com/dominictarr/rc/pull/72), created a year ago. Author hasn't responded yet. 

Anyways, seems to be a deliberate design decision, rather than something overlooked. It was confusing for me. Maybe worth adding a notice about this to the docs at least?

**Comment by springmeyer** [comment]
> Anyways, seems to be a deliberate design decision, rather than something overlooked. However this behaviour was confusing to me.

Okay, thanks for researching this back to https://github.com/dominictarr/rc/pull/72

> Maybe worth adding a notice about this to the README at least?

Definitely. Happy to accept a PR on this.

**Comment by szdavid92** [comment]
Created #282

**Comment by n0mer** [comment]
> The reason is that Windows treats them so, and the people are pretty used to this behaviour.

That is weak argument. 
Is that behavior POSIX-compliant?
